### PR TITLE
DOCS-2475: Add additional headers

### PR DESCRIPTION
**Pull request by npentrel**

### CODE REVIEW

File: docs/build/configure/triggers.md [hunk 1]
```diff
@@ -252,10 +295,13 @@ The request includes the following headers:
 <!-- prettier-ignore -->
 | Header Key | Description |
 | ---------- | ----------- |
-| `Part-ID` |  The part of the machine that triggered the request. |
-| `Robot-ID` | The machine that triggered the request. |
-| `Location-ID` | The location of the machine that triggered the request. |
-| `Org-ID` | The organization that triggered the request. |
+| `Org-Id` | The ID of the organization that triggered the request. |
```
Comment (npentrel) [inline docs/build/configure/triggers.md:298]: @maxhorowitz : I've not been able to confirm these all work:

Here is an example of a trigger from ingested binary data. I don't get the associated component type, name, method name, and min-time received, max time received.
```
127.0.0.1 - - [18/Jul/2024 17:49:42] "POST /?data_type=binary&event_type=part_data_ingested HTTP/1.1" 200 -
ImmutableMultiDict([('data_type', 'binary'), ('event_type', 'part_data_ingested')])
Host: d99b-71-167-222-76.ngrok-free.app
User-Agent: Go-http-client/2.0
Content-Length: 375
Accept-Encoding: gzip
Content-Type: application/json
Location-Id: vw3iu72d8n
Org-Id: bccf8f8f-e3c4-4f72-ab9a-fc547757f352
Part-Id: 422afb68-21d0-487c-a135-01d81ff54407
Robot-Id: 584b3120-0da5-4014-b843-8a685e9cab2d
X-Forwarded-For: 2600:1900:2000:a3::1:900
X-Forwarded-Host: d99b-71-167-222-76.ngrok-free.app
X-Forwarded-Proto: https
```

Comment (npentrel) [reply docs/build/configure/triggers.md:298]: And then here's an example of `part_online`:

```
127.0.0.1 - - [18/Jul/2024 17:48:12] "GET /?event_type=part_online HTTP/1.1" 200 -
ImmutableMultiDict([('data_type', 'binary'), ('event_type', 'part_data_ingested')])
Host: d99b-71-167-222-76.ngrok-free.app
User-Agent: Go-http-client/2.0
Content-Length: 375
Accept-Encoding: gzip
Content-Type: application/json
Location-Id: vw3iu72d8n
Org-Id: bccf8f8f-e3c4-4f72-ab9a-fc547757f352
Part-Id: 422afb68-21d0-487c-a135-01d81ff54407
Robot-Id: 584b3120-0da5-4014-b843-8a685e9cab2d
X-Forwarded-For: 107.178.207.105
X-Forwarded-Host: d99b-71-167-222-76.ngrok-free.app
X-Forwarded-Proto: https
```

File: docs/build/configure/triggers.md [hunk 2]
```diff
@@ -250,17 +293,20 @@ When a trigger occurs, Viam sends a HTTP request to the URL you specified for th
...
-| `Org-ID` | The organization that triggered the request. |
-| `Component-Type` | The type of component for which data was ingested. Only for `part_data_ingested` triggers. |
-| `Component-Name` | The name of the component for which data was ingested. Only for `part_data_ingested` triggers. |
-| `Method-Name` | The name of the method from which data was ingested. Only for `part_data_ingested` triggers. |
-| `Min-Time-Received` | Indicates the earliest time a piece of data was received. |
-| `Max-Time-Received` | Indicates the latest time a piece of data was received. |
+| Header Key | Description | Trigger types |
+| ---------- | ----------- | ------------- |
+| `Org-Id` | The ID of the organization that triggered the request. | `part_data_ingested` |
+| `Organization-Name` | The name of the organization that triggered the request. | `part_online`, `part_offline` |
```
Comment (npentrel) [inline docs/build/configure/triggers.md:299]: @maxhorowitz also why is this so awkward? Shouldn't they all return org-id, org name, location id, location name, etc?

File: docs/build/configure/triggers.md (hunk 1 above)
Comment (maxhorowitz) [reply docs/build/configure/triggers.md]: ## 1st message -
component type, name, method name, min-time received, and max time received were moved to the data ingested POST request body - I am sorry if I grazed over this in a previous review - it was a change that was made during the introduction of POST request for part data ingested webhooks
Screenshot 2024-07-22 at 4 35 59 PM

## 2nd message -
https://viam.atlassian.net/browse/APP-5608

File: docs/build/configure/triggers.md (hunk 2 above)
Comment (maxhorowitz) [reply docs/build/configure/triggers.md:303]: TLDR is this is a bug - ticket is [here](https://viam.atlassian.net/browse/APP-5608) - is this something you'd like prio'd ?

File: docs/build/configure/triggers.md [hunk 3]
```diff
@@ -209,32 +209,79 @@ The following attributes are available for triggers:
...
+   {{% tab name="Flask" %}}
+
+```python {class="line-numbers linkable-line-numbers" }
+from flask import Flask, request
+
+app = Flask(__name__)
+
+
+@app.route("/", methods=['GET', 'POST'])
+def hello_http_get():
```
Comment (maxhorowitz) [inline docs/build/configure/triggers.md:222]: add post too?

File: docs/build/configure/triggers.md [hunk 4]
```diff
@@ -209,32 +209,79 @@ The following attributes are available for triggers:
...
+        "Part-Id": headers.get('part-id', 'no value'),
+        "Part-Name": headers.get('part-name', 'no value'),
+        "Robot-Id": headers.get('robot-id', 'no value'),
+        "Machine-Name": headers.get('machine-name', 'no value'),
+        "Component-Type": headers.get('component-type', 'no value'),
+        "Component-Name": headers.get('component-name', 'no value'),
+        "Method-Name": headers.get('method-name', 'no value'),
+        "Min-Time-Received": headers.get('min-time-received', 'no value'),
+        "Max-Time-Received": headers.get('max-time-received', 'no value'),
+        "Data-Type": request.args.get('data_type', 'no value')
```
Comment (maxhorowitz) [inline docs/build/configure/triggers.md:239]: ComponentName, ComponentType, MethodName, MachineName, LocationName, OrgName, FileID, FilePath are each parsed from within the request's body, not headers

File: docs/build/configure/triggers.md (hunk 3 above)
Comment (npentrel) [reply docs/build/configure/triggers.md:222]: Oh it'll do both but I can rename the function to not mention get 😅

File: docs/build/configure/triggers.md (hunk 2 above)
Comment (npentrel) [reply docs/build/configure/triggers.md:303]: No all good. Was just wondering if it had been an oversight. No urgency from my end

**Comment by maxhorowitz** [comment]
FYI closing https://viam.atlassian.net/browse/APP-5608 today 👍

File: docs/build/configure/triggers.md [hunk 5]
```diff
@@ -250,17 +308,30 @@ When a trigger occurs, Viam sends a HTTP request to the URL you specified for th
...
-| `Part-ID` |  The part of the machine that triggered the request. |
-| `Robot-ID` | The machine that triggered the request. |
-| `Location-ID` | The location of the machine that triggered the request. |
-| `Org-ID` | The organization that triggered the request. |
-| `Component-Type` | The type of component for which data was ingested. Only for `part_data_ingested` triggers. |
-| `Component-Name` | The name of the component for which data was ingested. Only for `part_data_ingested` triggers. |
-| `Method-Name` | The name of the method from which data was ingested. Only for `part_data_ingested` triggers. |
-| `Min-Time-Received` | Indicates the earliest time a piece of data was received. |
-| `Max-Time-Received` | Indicates the latest time a piece of data was received. |
+| Header Key | Description | Trigger types |
```
Comment (npentrel) [inline docs/build/configure/triggers.md:311]: @maxhorowitz are these the correct trigger types for each header key?

File: docs/build/configure/triggers.md [hunk 6]
```diff
@@ -209,32 +209,90 @@ The following attributes are available for triggers:
...
+        "Organization-Name": headers.get('organization-name', '') or
+                             data.get('org_name', 'no value'),
+        "Location-Id": headers.get('location-id', 'no value'),
+        "Location-Name": headers.get('location-name', '') or
+                         data.get('location_name', 'no value'),
+        "Part-Id": headers.get('part-id', 'no value'),
+        "Part-Name": headers.get('part-name', 'no value'),
+        "Robot-Id": headers.get('robot-id', 'no value'),
+        "Machine-Name": headers.get('machine-name', '') or
+                        data.get('machine_name', 'no value'),
```
Comment (npentrel) [inline docs/build/configure/triggers.md:238]:
```suggestion
        "Org-Id": headers.get('org-id', 'no value'),
        "Organization-Name": headers.get('organization-name', '') or
         data.get('org_name', 'no value'),
        "Location-Id": headers.get('location-id', 'no value'),
        "Location-Name": headers.get('location-name', '') or
         data.get('location_name', 'no value'),
        "Part-Id": headers.get('part-id', 'no value'),
        "Part-Name": headers.get('part-name', 'no value'),
        "Robot-Id": headers.get('robot-id', 'no value'),
        "Machine-Name": headers.get('machine-name', '') or
         data.get('machine_name', 'no value'),
```

**Review by maxhorowitz — APPROVED** [review]
Cross checked with the code and everything seems correct to me - as long as you've manually tested that lambda against a trigger we're good to merge! Thank you

**Comment by npentrel** [comment]
yep all tested

**Pull request merged**